cmake/OpenCVFindLibsGUI.cmake

```diff
@@ -69,6 +69,7 @@ if(WITH_OPENGL)
       if(QT_QTOPENGL_FOUND)
         set(HAVE_QT_OPENGL TRUE)
       else()
+        list(APPEND OPENCV_LINKER_LIBS ${OPENGL_LIBRARIES})
```

Comment:
Global `OPENCV_LINKER_LIBS` is going to be deprecated. This is why this line is removed. Please check #21213.
We need to add `ocv_target_link_libraries(${the_module} PRIVATE "${OPENGL_LIBRARIES}")` somewhere in highgui's CMakeLists under some conditions (probably `if(HAVE_OPENGL)`).